pkg/db/db.go

```diff
@@ -174,24 +174,33 @@ func (db *DB) SelectIndexByArtifactIDAndGroupID(artifactID, groupID string) (typ
 	return index, nil
 }
 
-func (db *DB) SelectIndexesByArtifactIDAndFileType(artifactID string, fileType types.ArchiveType) ([]types.Index,
+func (db *DB) SelectGroupIDByArtifactIDVersionAndFileType(artifactID, version string, fileType types.ArchiveType) (string,
 	error) {
-	var indexes []types.Index
+	var groupID string
+	var count int
 	rows, err := db.client.Query(`
-		SELECT a.group_id, a.artifact_id, i.version, i.sha1, i.archive_type
-		FROM indices i 
-		JOIN artifacts a ON a.id = i.artifact_id
-        WHERE a.artifact_id = ? AND i.archive_type = ?`,
-		artifactID, fileType)
+		SELECT relevant.group_id, COUNT(relevant.group_id) as count
+		FROM indices i
+		JOIN (SELECT a.id, a.group_id
+        	  FROM indices i
+        	  JOIN artifacts a on a.id = i.artifact_id
+        	  WHERE a.artifact_id = ? AND i.version = ? AND i.archive_type = ?) relevant ON relevant.id = i.artifact_id
+		GROUP BY "group_id"`,
+		artifactID, version, fileType)
```

Comment:
If I understand correctly, we need to add one more `SELECT` to get maximum value from `count` in sql query.
I'm not sure we need this.

I simply added logic to select the maximum quantity in `rows.Next()` loop.